fix pathFilter for alt browser field

mvayngrib opened the pull request:
oops, missed this one the first time around. Browserify's tests unearthed it

defunctzombie commented on index.js:
```diff
@@ -180,17 +180,17 @@ function build_resolve_opts(opts, base) {
     	}
-    	if(!info.browser) {
+
+        var replacements = info[browser];
```
syntax isssue

defunctzombie commented on index.js:
```diff
@@ -180,17 +180,17 @@ function build_resolve_opts(opts, base) {
-    		return mappedPath;
-    	}
+		mappedPath = replacements[relativePath];
+		if(!mappedPath && (relativePath.lastIndexOf(".js") === relativePath.length-3) ) {
```
single quotes for strings

defunctzombie commented on index.js:
```diff
@@ -180,17 +180,17 @@ function build_resolve_opts(opts, base) {
-    	}
+		mappedPath = replacements[relativePath];
+		if(!mappedPath && (relativePath.lastIndexOf(".js") === relativePath.length-3) ) {
+			mappedPath = replacements[relativePath+".js"];
```
single quotes for strings, space around `+` operator

mvayngrib replied:
np, thought it was your style choice

defunctzombie replied:
maybe an oversight on a previous PR or laziness on my part.

defunctzombie commented:
Please add tests so it is not broken again; then I will merge.

mvayngrib commented:
added missing tests. "deep module reference mapping (alt-browser)" detects the overlooked pathFilter bug.

also some style fixes, and added .jscsrc, idk if you want that in here

defunctzombie commented:
What's with all the changes to the test fixtures?

mvayngrib commented:
added "chromeapp" fields to various package.json's, added files "chromeapp" fields pull in, to differentiate from the ones pulled in by "browser" fields

mvayngrib commented:
i'm starting to think separate tests for an alt browser field are unnecessary, so I wouldn't merge this if I were you. In fact the previous tests I added should be scrapped too, as "browser" is now nothing more than a default.

defunctzombie commented:
K, so can we make just a test for the issue this is fixing?

On Friday, March 27, 2015, Mark Vayngrib <notifications@github.com> wrote:

> i'm starting to think separate tests for an alt browser field are
> unnecessary, so I wouldn't merge this if I were you. In fact the previous
> tests I added should be scrapped too, as "browser" is now nothing more than
> a default.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/defunctzombie/node-browser-resolve/pull/58#issuecomment-87131805>
> .
>

mvayngrib commented:
added, my bad dragging this out

defunctzombie commented:
Thanks for being vigilant and following up! Fix is out in 1.8.2
